lib/search_click_tracking/service.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'uri'
+require 'common/client/base'
+require 'common/client/concerns/monitoring'
+require 'common/client/errors'
+require 'search_click_tracking/response'
+require 'search_click_tracking/configuration'
+
+module SearchClickTracking
+  # This class builds a wrapper around Search.gov web results API. Creating a new instance of class
+  # will and calling #results will return a ResultsResponse upon success or an exception upon failure.
+  #
+  # @see https://search.usa.gov/sites/7378/api_instructions
+  #
+  class Service < Common::Client::Base
+    include Common::Client::Concerns::Monitoring
+
+    STATSD_KEY_PREFIX = 'api.search_click_tracking'
+
+    configuration SearchClickTracking::Configuration
+
+    attr_reader :url
+    attr_reader :query
+    attr_reader :position
+    attr_reader :client_ip
+    attr_reader :user_agent
+
+    def initialize(url, query, position, client_ip, user_agent)
+      @url = url
+      @query = query
+      @position = position
+      @client_ip = client_ip
+      @user_agent = user_agent
+    end
+
+    # POSTs click tracking query param data to search.gov 
+    #
+    def track_click
+      with_monitoring do
+        response = perform(:post, track_click_url, query_params, {'Content-Type' => 'application/x-www-form-urlencoded'})
```

Comment:
`query_params` is `"affiliate=va&access_key=TESTKEY&module_code=I14Y&url=&query=&position=&client_ip=&user_agent="` here, and the error seems to be the response isn't a json hash but `"Query can't be blank"`